### PR TITLE
refactor(meta): refactor compaction task sending and cancellation.

### PR DESCRIPTION
**Pull request by zwang28**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

Refactor as mentioned in https://github.com/singularity-data/risingwave/pull/4496#issuecomment-1209290724

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* Installation and deployment 
* Connector (sources & sinks)
* SQL commands, functions, and operators
* RisingWave cluster configuration changes
* Other (please specify in the release note below)

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

### CODE REVIEW

File: src/meta/src/hummock/compaction_scheduler.rs [hunk 1]
```diff
@@ -214,6 +199,36 @@ where
...
+                    e
+                );
+                // Cancel the task at best effort
+                compact_task.task_status = false;
+                if let Err(e) = self
+                    .hummock_manager
+                    .report_compact_task(compactor.context_id(), &compact_task)
+                    .await
+                {
+                    tracing::warn!("Failed to cancel task {}. {:#?}", compact_task.task_id, e);
```
Comment (Little-Wallace) [inline src/meta/src/hummock/compaction_scheduler.rs:221]: I think it shall be `error!` log.
The only fail reason is caused by IO (meta-storage).

Comment (zwang28) [reply src/meta/src/hummock/compaction_scheduler.rs]: fixed

File: src/meta/src/hummock/compaction_scheduler.rs [hunk 2]
```diff
@@ -214,6 +199,36 @@ where
...
+                .await
+            {
+                tracing::warn!(
+                    "Failed to send task {} to {}. {:#?}",
+                    compact_task.task_id,
+                    compactor.context_id(),
+                    e
+                );
+                // Cancel the task at best effort
+                compact_task.task_status = false;
```
Comment (Li0k) [inline src/meta/src/hummock/compaction_scheduler.rs:215]: `cancle_compaction` logic similar with `need_cancel_task` in `hummock/manager/mod.rs`, shall we provide  a unified function ?

Rest LGTM ！ thanks

**Review by Li0k — APPROVED** [review]

File: src/meta/src/hummock/compaction_scheduler.rs (hunk 2 above)
Comment (zwang28) [reply src/meta/src/hummock/compaction_scheduler.rs:215]: Agree. But let me keep it straightforward for now, to reduce potential conflict with another RP (reliable task cancellation).

**Comment by zwang28** [comment]
@mergifyio requeue

**Pull request merged**